flink-libraries/flink-gelly-examples/src/test/java/org/apache/flink/graph/drivers/DriverBaseITCase.java

```diff
@@ -41,17 +43,41 @@
 	@Rule
 	public ExpectedException expectedException = ExpectedException.none();
 
-	protected DriverBaseITCase(TestExecutionMode mode) {
+	protected final String idType;
+
+	protected DriverBaseITCase(TestExecutionMode mode, String idType) {
 		super(mode);
+
+		this.idType = idType;
 	}
 
-	// extend MultipleProgramsTestBase default to include object reuse mode
-	@Parameterized.Parameters(name = "Execution mode = {0}")
+	@Parameterized.Parameters(name = "Execution mode = {0}, ID type = {1}")
 	public static Collection<Object[]> executionModes() {
-		return Arrays.asList(
-			new Object[] { TestExecutionMode.CLUSTER },
-			new Object[] { TestExecutionMode.CLUSTER_OBJECT_REUSE },
-			new Object[] { TestExecutionMode.COLLECTION });
+		List<Object[]> executionModes = new ArrayList<>();
+
+		for (TestExecutionMode executionMode : TestExecutionMode.values()) {
+			for (String idType : new String[] {"byte", "nativeByte", "short", "nativeShort", "char", "nativeChar",
+									"integer", "nativeInteger", "long", "nativeLong", "string", "nativeString"}) {
```

Comment:
What do you think if we declare the list of `idType` before the outer loop?
IMHO, this will make formation better.